### PR TITLE
Add kubeletconfig round trip test

### LINKED ISSUE #55959: kubeletconfig API group is missing a fuzzer round trip test
Solution: I will add one.

### PR DESCRIPTION
**Pull request by mtaufen**
I noticed we were missing one of these.

fixes #55959

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/apis/kubeletconfig/fuzzer/fuzzer.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	"k8s.io/kubernetes/pkg/kubelet/apis/kubeletconfig/v1alpha1"
+	"k8s.io/kubernetes/pkg/kubelet/qos"
+	kubetypes "k8s.io/kubernetes/pkg/kubelet/types"
+	"k8s.io/kubernetes/pkg/master/ports"
+)
+
+// Funcs returns the fuzzer functions for the kubeadm apis.
+func Funcs(codecs runtimeserializer.CodecFactory) []interface{} {
+	return []interface{}{
+		// Match the defaulter, so there are no empty values for it to change during the roundtrip
```
Comment (yujuhong) [inline pkg/kubelet/apis/kubeletconfig/fuzzer/fuzzer.go]: Why do you have to match the defaulter?

Comment (luxas) [reply pkg/kubelet/apis/kubeletconfig/fuzzer/fuzzer.go]: I don't _think_ this **necessarily** have to match the defaulter, but most probably works if you do so (as it leaves no fields empty)

File: pkg/kubelet/apis/kubeletconfig/fuzzer/fuzzer.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeserializer "k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/kubernetes/pkg/kubelet/apis/kubeletconfig"
+	"k8s.io/kubernetes/pkg/kubelet/apis/kubeletconfig/v1alpha1"
+	"k8s.io/kubernetes/pkg/kubelet/qos"
+	kubetypes "k8s.io/kubernetes/pkg/kubelet/types"
+	"k8s.io/kubernetes/pkg/master/ports"
+)
+
+// Funcs returns the fuzzer functions for the kubeadm apis.
```
Comment (luxas) [inline pkg/kubelet/apis/kubeletconfig/fuzzer/fuzzer.go]: We don't have to point out kubeadm here, but consumers of this API generally most probably like to use the roundtrip tests where this makes it much easier to consume

**Review by luxas — APPROVED** [review]
Thanks for this :+1:!
Great to see this new coverage for the soon-to-be-beta API group!

For the record, this will probably help us over in https://github.com/kubernetes/kubernetes/pull/55803 where we consume these API types embedded in an other kubeadm type that already has fuzzing/roundtrip tests enabled.

File: pkg/kubelet/apis/kubeletconfig/fuzzer/fuzzer.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+func Funcs(codecs runtimeserializer.CodecFactory) []interface{} {
```
Comment (xiangpengzhao) [inline pkg/kubelet/apis/kubeletconfig/fuzzer/fuzzer.go]: @mtaufen I guess you need to register `kubeletconfigfuzzer.Funcs`

https://github.com/kubernetes/kubernetes/blob/dd9981d038012c120525c9e6df98b3beb3ef19e1/pkg/api/testing/fuzzer.go#L92-L107

File: pkg/kubelet/apis/kubeletconfig/fuzzer/fuzzer.go (hunk 1 above)
Comment (mtaufen) [reply pkg/kubelet/apis/kubeletconfig/fuzzer/fuzzer.go]: You don't have to match the defaulter. It was just the easiest way to get valid non-empty values so the defaulter wouldn't mess up the round trip test. I'll clarify the comment.

File: pkg/kubelet/apis/kubeletconfig/fuzzer/fuzzer.go (hunk 2 above)
Comment (mtaufen) [reply pkg/kubelet/apis/kubeletconfig/fuzzer/fuzzer.go]: Oh whoops, copypasta error :D

File: pkg/kubelet/apis/kubeletconfig/fuzzer/fuzzer.go (hunk 3 above)
Comment (mtaufen) [reply pkg/kubelet/apis/kubeletconfig/fuzzer/fuzzer.go]: What does registering fuzzer funcs here do, vs just passing them into a call to RoundTripTestForAPIGroup?

Comment (xiangpengzhao) [reply pkg/kubelet/apis/kubeletconfig/fuzzer/fuzzer.go]: I'm not sure about it. Just seeing other fuzzers register here so I guess the newly added fuzzer also needs to register :)

Comment (xiangpengzhao) [reply pkg/kubelet/apis/kubeletconfig/fuzzer/fuzzer.go]: I tested locally, whether it is registered here or not, the test can pass...
@sttts may have some contexts. Thanks!

Comment (sttts) [reply pkg/kubelet/apis/kubeletconfig/fuzzer/fuzzer.go]: Please only add fuzzer for apiserver types here, not for configs only used locally. Look for roundtrip_test.go files in the code-base. There are some more which have local fuzzers.

Comment (xiangpengzhao) [reply pkg/kubelet/apis/kubeletconfig/fuzzer/fuzzer.go]: @sttts thanks for clarifying!

**Comment by jberkus** [comment]
Hey, it's Code Slush today.  If it's possible to mark this PR approved and approved-for-milestone that will make 1.9 merge easier.  Thanks!

**Comment by jiayingz** [comment]
@mtaufen 
Do you think the recent submit queue pull-kubernetes-bazel-test failure could be related to this pr?
https://prow.k8s.io/?type=batch
Example failure: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/batch/pull-kubernetes-bazel-test/18558/

**Comment by mtaufen** [comment]
This probably needs to be rebased. Holding because the bot didn't add a needs-rebase label.

**Comment by mtaufen** [comment]
Rebased

**Comment by mtaufen** [comment]
Re-adding label (issue was that a struct used in a brand-new file in this PR changed upstream, but this wouldn't cause a merge conflict).

**Comment by mtaufen** [comment]
Thanks @jiayingz for noticing!

**Pull request merged**